### ISSUE TITLE
Crashing on API 19

### ISSUE THREAD

**Issue by joaolisboa** [centic9/poi-on-android#12]
A bunch of errors appear related to the ooxml.schemas, examples:

Could not find class 'org.openxmlformats.schemas.spreadsheetml.x2006.main.CTExtensionList', referenced from method org.openxmlformats.schemas.spreadsheetml.x2006.main.impl.CTSstImpl.getExtLst

Could not find class 'org.openxmlformats.schemas.spreadsheetml.x2006.main.impl.CTSstImpl$1SiList', referenced from method org.openxmlformats.schemas.spreadsheetml.x2006.main.impl.CTSstImpl.getSiList

Related to the actual crash:

`FATAL EXCEPTION: main
                                                                    Process: ipleiria.project.add, PID: 6186
                                                                    java.lang.IllegalStateException: org.apache.poi.POIXMLException: java.lang.reflect.InvocationTargetException
                                                                        at ipleiria.project.add.utils.FileUtils.readExcel(FileUtils.java:182)
                                                                        at ipleiria.project.add.view.categories.CategoriesPresenter$1$1.onComplete(CategoriesPresenter.java:70)
                                                                        at ipleiria.project.add.view.categories.CategoriesPresenter$1$1.onComplete(CategoriesPresenter.java:67)
                                                                        at ipleiria.project.add.data.source.database.ItemsRepository$1.onDataChange(ItemsRepository.java:115)
                                                                        at com.google.firebase.database.Query$1.onDataChange(Unknown Source)
                                                                        at com.google.android.gms.internal.zzbpx.zza(Unknown Source)
                                                                        at com.google.android.gms.internal.zzbqx.zzZV(Unknown Source)
                                                                        at com.google.android.gms.internal.zzbra$1.run(Unknown Source)
                                                                        at android.os.Handler.handleCallback(Handler.java:733)
                                                                        at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                        at android.os.Looper.loop(Looper.java:136)
                                                                        at android.app.ActivityThread.main(ActivityThread.java:5017)
                                                                        at java.lang.reflect.Method.invokeNative(Native Method)
                                                                        at java.lang.reflect.Method.invoke(Method.java:515)
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779)
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595)
                                                                        at dalvik.system.NativeStart.main(Native Method)
                                                                     Caused by: org.apache.poi.POIXMLException: java.lang.reflect.InvocationTargetException
                                                                        at org.apache.poi.POIXMLFactory.createDocumentPart(POIXMLFactory.java:63)
                                                                        at org.apache.poi.POIXMLDocumentPart.read(POIXMLDocumentPart.java:604)
                                                                        at org.apache.poi.POIXMLDocument.load(POIXMLDocument.java:186)
                                                                        at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:296)
                                                                        at ipleiria.project.add.utils.FileUtils.readExcel(FileUtils.java:138)
                                                                        at ipleiria.project.add.view.categories.CategoriesPresenter$1$1.onComplete(CategoriesPresenter.java:70) 
                                                                        at ipleiria.project.add.view.categories.CategoriesPresenter$1$1.onComplete(CategoriesPresenter.java:67) 
                                                                        at ipleiria.project.add.data.source.database.ItemsRepository$1.onDataChange(ItemsRepository.java:115) 
                                                                        at com.google.firebase.database.Query$1.onDataChange(Unknown Source) 
                                                                        at com.google.android.gms.internal.zzbpx.zza(Unknown Source) 
                                                                        at com.google.android.gms.internal.zzbqx.zzZV(Unknown Source) 
                                                                        at com.google.android.gms.internal.zzbra$1.run(Unknown Source) 
                                                                        at android.os.Handler.handleCallback(Handler.java:733) 
                                                                        at android.os.Handler.dispatchMessage(Handler.java:95) 
                                                                        at android.os.Looper.loop(Looper.java:136) 
                                                                        at android.app.ActivityThread.main(ActivityThread.java:5017) 
                                                                        at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                        at java.lang.reflect.Method.invoke(Method.java:515) 
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779) 
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595) 
                                                                        at dalvik.system.NativeStart.main(Native Method) 
                                                                     Caused by: java.lang.reflect.InvocationTargetException
                                                                        at java.lang.reflect.Constructor.constructNative(Native Method)
                                                                        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
                                                                        at org.apache.poi.xssf.usermodel.XSSFFactory.createDocumentPart(XSSFFactory.java:56)
                                                                        at org.apache.poi.POIXMLFactory.createDocumentPart(POIXMLFactory.java:60)
                                                                        at org.apache.poi.POIXMLDocumentPart.read(POIXMLDocumentPart.java:604) 
                                                                        at org.apache.poi.POIXMLDocument.load(POIXMLDocument.java:186) 
                                                                        at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:296) 
                                                                        at ipleiria.project.add.utils.FileUtils.readExcel(FileUtils.java:138) 
                                                                        at ipleiria.project.add.view.categories.CategoriesPresenter$1$1.onComplete(CategoriesPresenter.java:70) 
                                                                        at ipleiria.project.add.view.categories.CategoriesPresenter$1$1.onComplete(CategoriesPresenter.java:67) 
                                                                        at ipleiria.project.add.data.source.database.ItemsRepository$1.onDataChange(ItemsRepository.java:115) 
                                                                        at com.google.firebase.database.Query$1.onDataChange(Unknown Source) 
                                                                        at com.google.android.gms.internal.zzbpx.zza(Unknown Source) 
                                                                        at com.google.android.gms.internal.zzbqx.zzZV(Unknown Source) 
                                                                        at com.google.android.gms.internal.zzbra$1.run(Unknown Source) 
                                                                        at android.os.Handler.handleCallback(Handler.java:733) 
                                                                        at android.os.Handler.dispatchMessage(Handler.java:95) 
                                                                        at android.os.Looper.loop(Looper.java:136) 
                                                                        at android.app.ActivityThread.main(ActivityThread.java:5017) 
                                                                        at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                        at java.lang.reflect.Method.invoke(Method.java:515) 
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779) 
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595) 
                                                                        at dalvik.system.NativeStart.main(Native Method) 
                                                                     Caused by: java.lang.VerifyError: org/openxmlformats/schemas/spreadsheetml/x2006/main/impl/CTRstImpl
                                                                        at java.lang.reflect.Constructor.constructNative(Native Method)
                                                                        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
                                                                        at org.apache.xmlbeans.impl.schema.SchemaTypeImpl.createUnattachedNode(SchemaTypeImpl.java:1934)
                                                                        at org.apache.xmlbeans.impl.schema.SchemaTypeImpl.createElementType(SchemaTypeImpl.java:1051)
                                                                        at org.apache.xmlbeans.impl.values.XmlObjectBase.create_element_user(XmlObjectBase.java:938)
                                                                        at org.apache.xmlbeans.impl.store.Xobj.getUser(Xobj.java:1675)
                                                                        at org.apache.xmlbeans.impl.store.Xobj.find_all_element_users(Xobj.java:2098)
                                                                        at org.openxmlformats.schemas.spreadsheetml.x2006.main.impl.CTSstImpl.getSiArray(Unknown Source)
                                                                        at org.apache.poi.xssf.model.SharedStringsTable.readFrom(SharedStringsTable.java:123)
                                                                        at org.apache.poi.xssf.model.SharedStringsTable.<init>(SharedStringsTable.java:107)
                                                                        at java.lang.reflect.Constructor.constructNative(Native Method) 
                                                                        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) 
                                                                        at org.apache.poi.xssf.usermodel.XSSFFactory.createDocumentPart(XSSFFactory.java:56) 
                                                                        at org.apache.poi.POIXMLFactory.createDocumentPart(POIXMLFactory.java:60) 
                                                                        at org.apache.poi.POIXMLDocumentPart.read(POIXMLDocumentPart.java:604) 
                                                                        at org.apache.poi.POIXMLDocument.load(POIXMLDocument.java:186) 
                                                                        at org.apache.poi.xssf.usermodel.XSSFWorkbook.<init>(XSSFWorkbook.java:296) 
                                                                        at ipleiria.project.add.utils.FileUtils.readExcel(FileUtils.java:138) 
                                                                        at ipleiria.project.add.view.categories.CategoriesPresenter$1$1.onComplete(CategoriesPresenter.java:70) 
                                                                        at ipleiria.project.add.view.categories.CategoriesPresenter$1$1.onComplete(CategoriesPresenter.java:67) 
                                                                        at ipleiria.project.add.data.source.database.ItemsRepository$1.onDataChange(ItemsRepository.java:115) 
                                                                        at com.google.firebase.database.Query$1.onDataChange(Unknown Source) 
                                                                        at com.google.android.gms.internal.zzbpx.zza(Unknown Source) 
                                                                        at com.google.android.gms.internal.zzbqx.zzZV(Unknown Source) 
                                                                        at com.google.android.gms.internal.zzbra$1.run(Unknown Source) 
                                                                        at android.os.Handler.handleCallback(Handler.java:733) 
                                                                        at android.os.Handler.dispatchMessage(Handler.java:95) 
                                                                        at android.os.Looper.loop(Looper.java:136) 
                                                                        at android.app.ActivityThread.main(ActivityThread.java:5017) 
                                                                        at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                        at java.lang.reflect.Method.invoke(Method.java:515) 
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779) 
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595) 
                                                                        at dalvik.system.NativeStart.main(Native Method) `

**Comment by centic9** [comment]
Does the test-app that is included in poi-on-android work?

**Comment by joaolisboa** [comment]
I haven't tried it out to be honest. I already had experience with Apache POI so I already knew how to use it, I just checked the code to see if you were doing anything special or different but nothing more. On 4.4(API 19) it always crashes on the document creation. I only found out later during testing. I was only capable of fixing it by using another project(AndroidReadXLSX) with an older version(3.10 ooxml-schemas, 3.12 poi) and only then did it work.
